### PR TITLE
f-restaurant-card@v0.10.0 - add offers

### PR DESCRIPTION
**Pull request by jamieomaguire**
v0.10.0
------------------------------
*December 07, 2021*

### Added
- Restaurant offers
- associated unit tests to check render logic

### Visual demo

Screenshot 2021-12-07 at 10 58 10

Screenshot 2021-12-07 at 10 56 48

Screenshot 2021-12-07 at 10 56 59

### CODE REVIEW

**Review by simonsandersje — DISMISSED** [review]

File: packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue [hunk 1]
```diff
@@ -126,6 +126,14 @@
             </component>
             <!-- END ERROR BOUNDARY -->
 
+            <!-- Offers -->
+            <icon-text
+                v-if="hasOffer"
+                data-test-id="restaurant-offer"
+                :text="offer"
+                :is-bold="true">
```
Comment (DamianMullins) [inline packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue]: No need to update but you can shorthand props with a value of `true` like this one to remove the assignment part e.g. `is-bold`.

File: packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue [hunk 2]
```diff
@@ -222,6 +238,9 @@ export default {
         hasImageTags () {
             return !!this.tags?.imageTags?.length;
         },
+        hasOffer () {
+            return !!this.offer?.length || false;
```
Comment (DamianMullins) [inline packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue]: Don't think you need the `|| false` part here ase the `!!` operators should do this already?

File: packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue (hunk 1 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue]: I had no idea you could do that! thanks, I'll update as it's neat!

File: packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue (hunk 2 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue]: Oh yeah, not quite sure why I added that

File: packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCard.v1.test.js [hunk 3]
```diff
@@ -236,4 +236,32 @@ describe('RestaurantCard.v1', () => {
...
+            };
+
+            // act
+            const wrapper = mount(RestaurantCardV1, { propsData });
+
+            // assert
+            expect(wrapper.find('[data-test-id="restaurant-offer"]').exists()).toBe(true);
+        });
+
+        it('does not display an offer if none exists', () => {
```
Comment (dawidchar) [inline packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCard.v1.test.js]: Worth checking undefined and an empty string?

**Review by dawidchar — DISMISSED** [review]

File: packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCard.v1.test.js (hunk 3 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCard.v1.test.js]: undefined, sure. empty string - not so concerned. I don't think it's a likely scenario and I'd prefer not to call a trim method on every restaurant offer

Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCard.v1.test.js]: sorry - misread as a string with just spaces. I will add an empty string one.

**Review by dawidchar — APPROVED** [review]

**Review by DamianMullins — APPROVED** [review]

**Pull request merged**